py/runtime.c

```diff
@@ -715,16 +714,37 @@ void mp_call_prepare_args_n_kw_var(bool have_self, size_t n_args_n_kw, const mp_
     uint args2_alloc;
     uint args2_len = 0;
 
+    // Try to get a hint for unpacked * args length
+    int list_len = 0;
```

Comment:
Now that this and `kw_dict_len` are signed, it's not clear to me that the calculations that involve these variables are guaranteed to be >= 0.... did you follow through all cases to check this?

Reply:
They are guaranteed to be >= 0. `list_len` can be -1 if and only if `n_args` > 0 and it is only used as `n_args + list_len`. Likewise, `kw_dict_len` can be -1 if and only if `n_kw` is > 0 and is only used as `n_kw + kw_dict_len`. Neither value can ever by less than -1 since 1 is subtracted from an unsigned value.

We could add a couple of `assert()`s to try to catch any future regressions if the code is changed.

Reply:
`list_len` could be -2 if there are two lists, right?  Or even more negative.  But it sounds like that will still be offset by `n_args`.

Same with `kw_dict_len`?

Maybe there needs to be a test that passes in a few empty lists and empty dicts?

Reply:
> Neither value can ever by less than -1 since 1 is subtracted from an unsigned value.

This statement is not accurate. You are correct. The -1 is in a for loop that loops `n_args` or `n_kw` times, so `list_len` and `kw_dict_len` can be at the smallest `-n_args` and `-kw_dict_len` respectively. But is is still mathematically provable that `n_args + list_len` and `n_kw + kw_dict_len` can never be < 0.

Reply:
I think the `int` should be `ssize_t`